source/agora/common/Ensure.d

```diff
@@ -47,14 +47,49 @@ private final class FormattedException : Exception
     {
         return this.buffer[0 .. this.end];
     }
+
+    /// Overrides `Throwable.toString` because it uses `msg`...
+    public override void toString (scope void delegate(in char[]) sink) const scope
+    {
+        import core.internal.string : unsignedToTempString;
+
+        char[20] tmpBuff = void;
+
+        sink(typeid(this).name);
+        sink("@"); sink(file);
+        sink("("); sink(unsignedToTempString(line, tmpBuff)); sink(")");
+
+        if (this.end > 0)
+        {
+            sink(": "); sink(this.message());
```

Comment:
is using a custom `message()` more efficient than using the `msg` in `Throwable`?

Reply:
No, because `message` is a (virtual) function call while `msg` is a field.
But `message` was introduced for users that needed something like this (buffer + format).
Unfortunately, `toString` wasn't changed to use it.

Reply:
FYI: https://github.com/dlang/druntime/commit/e7e49cd206dec5324f9f4baf56099e211d8f9c9e